march_hardware/src/Joint.cpp

```diff
@@ -23,30 +24,24 @@ void Joint::initialize(int ecatCycleTime)
 
 void Joint::prepareActuation()
 {
-  ROS_ASSERT_MSG(this->getActuationMode() != ActuationMode::unknown, "The mode of actuation for joint %s is: %s",
-                 this->name.c_str(), this->getActuationMode().toString().c_str());
-  if (this->allowActuation)
+  if (!this->allowActuation)
   {
-    ROS_INFO("[%s] Preparing for actuation", this->name.c_str());
-    this->iMotionCube.goToOperationEnabled();
-    ROS_INFO("[%s] Successfully prepared for actuation", this->name.c_str());
-  }
-  else
-  {
-    ROS_ERROR("[%s] Trying to prepare for actuation while it is not "
-              "allowed to actuate",
-              this->name.c_str());
+    throw error::HardwareException(error::ErrorType::NOT_ALLOWED_TO_ACTUATE, "Trying to prepare joint %s for actuation",
```

Comment:
The message suggests that the program is still trying to prepare the joint for actuation, which is not the case.
```suggestion
    throw error::HardwareException(error::ErrorType::NOT_ALLOWED_TO_ACTUATE, "Failed to prepare joint %s for actuation",
```